src/fennel/specials.fnl

```diff
@@ -946,6 +946,22 @@ Only works in Lua 5.3+ or LuaJIT with the --use-bit-lib flag.")
         [rhs] (compiler.compile1 rhs-ast scope parent {:nval 1})]
     (string.format "(%s %s %s)" (tostring lhs) op (tostring rhs))))
 
+(fn idempotent-comparator [op chain-op ast scope parent]
+  "Compile a multi-arity comparison to a binary Lua comparison. Optimized
+  variant for values not at risk of double-eval."
+  (let [comparisons []
+        vals []
+        chain (string.format " %s " (or chain-op :and))]
+    (for [i 2 (length ast)]
```

Comment:
This would actually be better as `fcollect` instead of `for` but it occurs to me that we haven't backported that macro to the bootstrap compiler yet. Let me see if I can do that, then maybe you can rebase and switch this up?

Reply:
I noticed you did your end, so I did as suggested. Using `fcollect` in [fe3020f](https://github.com/bakpakin/Fennel/pull/439/commits/fe3020fc9143db10fdfecd9b64386b13c7628c68).